tests/test_diffraction.py

```diff
@@ -81,82 +81,79 @@ def test_dist_circle() -> None:
+            [-frac_sqrt_2, 0.0, frac_sqrt_2],
+        ]
     )
+    np.testing.assert_array_almost_equal(e_z(t), expected_e_z)
+    np.testing.assert_array_almost_equal(e_z_0, expected_e_z[1])
 
 
-def rot_2x2(alpha: np.array) -> np.array:
+def rot_2x2(alpha: np.ndarray) -> np.ndarray:
     rot = np.empty(np.shape(alpha) + (2, 2))
     rot[..., 0, 0] = rot[..., 1, 1] = np.cos(alpha)
-    rot[..., 0, 1] = np.sin(alpha)
-    rot[..., 1, 0] = -rot[..., 0, 1]
+    rot[..., 1, 0] = np.sin(alpha)
+    rot[..., 0, 1] = -rot[..., 1, 0]
     return rot
 
 
 def test_field_rotation_matrix_is_correct_at_ncp() -> None:
     """A telescope centered at the north celestial pole (NCP)
     will observe a uniform rotation of the stars around NCP."""
-    lat = 40.0
+    lat = 40.0 / 180.0 * np.pi
     t = 3600.0 * np.linspace(-2.0, 2.0, num=10)
-    e_star = np.array(
-        [[0.0, np.cos(lat / 180.0 * np.pi), np.sin(lat / 180.0 * np.pi)]] * t.size
-    )  # NCP
-    rot = diffraction.field_rotation_matrix(lat, e_star, t)
+    e_focal = np.array([0.0, 0.0, 1.0])  # NCP
+    e_z_0, e_z = diffraction.prepare_e_z(lat)
+    rot = diffraction.field_rotation_matrix(e_z_0, e_z, e_focal, t)
 
     # Around NCP, the field rotation angle should agree with earth's rotation angle
-    # around its axis:
-    np.testing.assert_array_almost_equal(rot, rot_2x2(diffraction.OMEGA_EARTH * t))
+    # around its axis (opposite sign):
+    np.testing.assert_array_almost_equal(rot, rot_2x2(-diffraction.OMEGA_EARTH * t))
 
 
 def test_field_rotation_matrix_is_correct_at_horizon_east_and_west() -> None:
-    """A star moving along the celestial equatorial will rise and set at east and west at the horizon."""
-    lat = 30.0
+    """A star moving along the celestial equator will rise and set at east and west at the horizon."""
+    lat = 30.0 / 180.0 * np.pi
     t = np.array([-T_SIDERIAL / 4.0, T_SIDERIAL / 4.0])
-    e_star = np.array([[-1.0, 0.0, 0.0], [1.0, 0.0, 0.0]])  # West and East at horizon
-    rot = diffraction.field_rotation_matrix(lat, e_star, t)
+    e_focal = np.array([1.0, 0.0, 0.0])  # Direction to celestial equator
+    e_z_0, e_z = diffraction.prepare_e_z(lat)
+    rot = diffraction.field_rotation_matrix(e_z_0, e_z, e_focal, t)
 
     # At horizon, the field rotation angle should be exactly +/- (90°-lat):
     np.testing.assert_array_almost_equal(
         rot,
-        np.array(
-            (
-                rot_2x2(-(90.0 - lat) / 180.0 * np.pi),
-                rot_2x2((90.0 - lat) / 180.0 * np.pi),
-            )
-        ),
+        rot_2x2(np.array([lat - 0.5 * np.pi, 0.5 * np.pi - lat])),
     )
 
 
 def test_field_rotation_matrix_is_correct_near_zenith() -> None:
     """Test if a star near zenith has a field rotation angle similar
     to the angle predicted by the field rotation rate formula."""
 
-    alt = 89.9
-    az = 45.0
-    lat = -30.24463
+    alt = 89.9 / 180.0 * np.pi
+    az = 45.0 / 180.0 * np.pi
+    lat = -30.24463 / 180.0 * np.pi
```

Comment:
Can we switch to using `coord.Angle` instances for these rather than passing floats around with implicit units?
So here `alt = 89.9 * coord.degrees`.  Then when using them, `np.cos(alt)` just works, or you can get the radian version if you need it with `alt.rad`.